server/plugin.go

```diff
@@ -484,6 +482,24 @@ func (p *Plugin) onActivate() error {
 		return err
 	}
 
+	p.remoteID, err = p.API.RegisterPluginForSharedChannels(model.RegisterPluginOpts{
+		Displayname:  pluginID,
+		PluginID:     pluginID,
+		CreatorID:    p.userID,
+		AutoShareDMs: true,
+		AutoInvited:  true,
+	})
+	if err != nil {
+		return err
+	}
+	p.API.LogInfo("Registered plugin for shared channels", "remote_id", p.remoteID)
+
+	if p.getConfiguration().DisableSyncMsg {
+		if err = p.API.UnregisterPluginForSharedChannels(pluginID); err != nil {
```

Comment:
Propose a log here to clarify that we did indeed disable it:
```
p.API.LogInfo("Unregistering plugin for shared channels since sync msg disabled")
```

Reply:
Good call